src/Algebra/Graph/AdjacencyMap/Extra.hs

```diff
@@ -0,0 +1,19 @@
+module Algebra.Graph.AdjacencyMap.Extra (
+  gtraverse,
+) where
+
+import Algebra.Graph.AdjacencyMap qualified as AM
+import Data.Set qualified as S
+
+-- | It's 'traverse', but for graphs
+--
+-- It's also unlawful. 'f' might be called several times for each node in the graph
+gtraverse ::
+  (Applicative f, Ord b) =>
+  (a -> f b) ->
+  AM.AdjacencyMap a ->
+  f (AM.AdjacencyMap b)
+gtraverse f = fmap mkAdjacencyMap . traverse (\(a, xs) -> (,) <$> f a <*> traverse f xs) . AM.adjacencyList
+  where
+    mkAdjacencyMap :: Ord c => [(c, [c])] -> AM.AdjacencyMap c
+    mkAdjacencyMap = AM.fromAdjacencySets . fmap (fmap S.fromList)
```

Comment:
[optional, readability]: I hate the `(,)` constructor so much, it's really annoying to read.  I usually use `tuplify`, since names are easier to read in prefix position than operators-in-parenthesis.
```suggestion
gtraverse f = fmap mkAdjacencyMap . traverse (\(a, xs) -> tuplify <$> f a <*> traverse f xs) . AM.adjacencyList
  where
    tuplify a b = (a, b)
    mkAdjacencyMap :: Ord c => [(c, [c])] -> AM.AdjacencyMap c
    mkAdjacencyMap = AM.fromAdjacencySets . fmap (fmap S.fromList)
```